src/zoom.ts

```diff
@@ -43,71 +44,43 @@ export async function getZoomInfo(): Promise<ZoomInfo> {
 	});
 
 	const active = meetingInfo.status === 'started';
-
-	if (!active) {
-		hasSeenStart = true;
-		participants = [];
-	}
-
-	if (!hasSeenStart) participants = [];
-
-	return { meetingInfo, active, hasSeenStart, participants };
-}
-
-export async function updateDiscordStatusFromZoom(discord: Client, zoomInfo?: ZoomInfo) {
-	if (!zoomInfo) zoomInfo = await getZoomInfo();
-
-	if (zoomInfo.active) {
-		// Curently, the outer ternary is redundant because status is only set when zoomInfo.active is true
-		const status = zoomInfo.active
-			? hasSeenStart && participants.length >= 0
-				? `with ${
-						participants.length === 1 ? `1 person` : `${participants.length} people`
-				  } on Zoom`
-				: 'on Zoom'
-			: 'with nobody'; /* currently disabled */
-
-		console.log(`Setting Discord status to: ${status}`);
-		await discord.user?.setActivity(status, { type: 'PLAYING' });
-	} else {
-		console.log('Clearing Discord Status');
-		// This is the best way I can find to clear activity (bots cant have custom statuses)
-		await discord.user?.setActivity('', { type: 'CUSTOM_STATUS' });
+	const wasActive = getCallIsActive(getState());
+	// status change
+	if (wasActive !== active) {
+		if (active) {
+			dispatch(callStarted);
+		} else {
+			dispatch(callEnded);
+		}
 	}
+	return meetingInfo.join_url;
```

Comment:
I strongly object to this method returning the join url, for a couple of reasons:
1. That's state, just like anything else, and should be store in redux appropriately.
2. This method has no relationship to returning the join url, so it's an arbitrary and confusing return value. (We could rename it to `updateZoomStatusAndGetJoinURL`, but that's obviously dumb.)
3. We don't need to reload all this information from Zoom every time we need the url. (It's more convenient if we reload it once per request and use a selector to get it multiple times fast.)